### DISCUSSION TITLE
(solved) Possible Implementation for: df robot mmWave - Human Presence Detection Sensor

### DISCUSSION THREAD

**Discussion by Roellen** [arendst/Tasmota#16411] [Q&A] [answered]
Hi. 
I bought this mmwave sensor which can detect human presence. 
Detection seems the best so far. But, as the sensor should send the "true" signal over MQTT as quick as possible I think I need a rule or so. 
I switched the teleperiod to 10 seconds, but it's way to slow. As lights in the room should be switched on directly. 

It's also possible to adjust settings over the TX/RX ports... such as detection length in meters, or the detection delay.
Not sure if it's possible to include this in tasmota so that we can change these values over MQTT. 
I can't do it. I can only solder wires, but no programming. 0:-)

I can send my mmwave sensor and wemos to someone for free, don't need it back - if someone want's to maybe add this mmWave sensor to tasmota.

https://www.dfrobot.com/product-2282.html
https://wiki.dfrobot.com/mmWave_Radar_Human_Presence_Detection_SKU_SEN0395

**Comment by barbudor** [comment, marked as answer]
What is the output of the sensor ? Is its a GPIO that change level when presence is detected ? 
If so you can read everything related to PIR sensor, different technology but same idea:
- Use a switch input (assuming `Switch_1`)
- [Detach from Relay](https://tasmota.github.io/docs/Buttons-and-Switches/#switchmode-15) (always good even if you don't have one) with `SwitchMode1 15`
- Tasmota publishes a SENSOR message immediatly on level change whatever is the teleperiod

et voilà 😄

**Reply by Roellen** [reply]
Whohooo!!!

It works perfectly! 
Actually very easy, if you know it. 🤪

Thanks a lot! 👌😬

The sensor HAS a high/low output. Easy to use.
I can’t change the settings over mqtt (i would like to change the detection radius) but… maybe this sensor will be fully implemented in the future. Who knows.

**Reply by sfromis** [reply]
The SEN0395 has both a digital switch-type output, and corresponding on/off signals available via serial. I'm already playing a bit with that specific sensor on ESP32, where I have a rudimentary Berry interface, including the ability to send commands to config up to 4 ranges, and delay/cool timing. What surprised me a bit is the ability to config 4 ranges, but still no output except for the very basic on/off, on serial data and the pin.

**Reply by Roellen** [reply]
I also bought the Aqara FP1 sensor. This one detects if you move to the sensor or from the sensor. (direction)
But… presence detection does not work 100% good. And detection is delayed like 3-5 seconds. So it needs to be combined with a PIR Sensor.

The only thing which looked promising to me was this:
https://youtu.be/VEqWlOeJ2YA

Hopefully it works as good as in the video.
